diff -Nru mingetty-0.9.4.orig/mingetty.c mingetty-0.9.4/mingetty.c
--- mingetty-0.9.4.orig/mingetty.c	Sun Apr 29 20:01:16 2001
+++ mingetty-0.9.4/mingetty.c	Sun Apr 29 20:04:59 2001
@@ -65,7 +65,7 @@
 
 /* AUTO_TTY is the tty on which autologins will be accepted.  If set
    to an empty string, autologins will be accepted on any tty. */
-#define AUTO_TTY "tty1"
+#define AUTO_TTY ""
 
 #include <sys/param.h>
 #define USE_SYSLOG
@@ -461,10 +461,15 @@
            a persistent record of the last time that an autologin
            request was granted.  Deny the autologin request if either
            the file open or file close fails. */
-	if ((fd=open(AUTO_LAST, O_WRONLY | O_CREAT | O_TRUNC, 0644)) < 0)
-		return 0;
-	if (close(fd) != 0)
-		return 0;
+
+   /* 
+    * bonkey: well, i need to autologin on 8 consoles, so don't
+    * bother me with such stuff :-> 
+    */
+//	if ((fd=open(AUTO_LAST, O_WRONLY | O_CREAT | O_TRUNC, 0644)) < 0)
+//		return 0;
+//	if (close(fd) != 0)
+//		return 0;
 
 	/* All tests are okay, so grant the autologin request. */
 	return 1;
