diff -urNp -x '*.orig' mingetty-1.08.org/mingetty.8 mingetty-1.08/mingetty.8
--- mingetty-1.08.org/mingetty.8	2021-05-09 20:16:02.173374609 +0200
+++ mingetty-1.08/mingetty.8	2021-05-09 20:16:02.210374609 +0200
@@ -6,7 +6,7 @@ mingetty \- minimal getty for consoles
 [\-\-noclear] [\-\-nonewline] [\-\-noissue] [\-\-nohangup] [\-\-nohostname]
 [\-\-long\-hostname] [\-\-loginprog=/bin/login] [\-\-nice=10] [\-\-delay=5]
 [\-\-chdir=/home] [\-\-chroot=/chroot] [\-\-autologin username]
-[\-\-loginpause] [\-\-mono]
+[\-\-loginpause] [\-\-mono] [\-\-defaultlogin username]
 .I tty
 .PP
 .SH DESCRIPTION
@@ -61,6 +61,9 @@ Change into this directory before callin
 .B \-\-chroot /chroot
 Call chroot() with this directory name.
 .TP
+.B \-\-defaultlogin username
+Log the specified user when no username is supplied in the prompt.
+.TP
 .B \-\-autologin username
 Log the specified user automatically in without asking for
 a login name and password. Check the \-f option from
diff -urNp -x '*.orig' mingetty-1.08.org/mingetty.c mingetty-1.08/mingetty.c
--- mingetty-1.08.org/mingetty.c	2021-05-09 20:16:02.173374609 +0200
+++ mingetty-1.08/mingetty.c	2021-05-09 20:16:02.210374609 +0200
@@ -98,6 +98,8 @@ static int loginpause = 0;
 static int mode = K_RAW;
 /* Set mono terminal type */
 static int mono_term = 0;
+/* If no login supplied, attempt a default login with this username. */
+static char *defaultlogin_name = NULL;
 
 /* error() - output error messages */
 static void error (const char *fmt, ...)
@@ -301,6 +303,27 @@ static void output_special_char (unsigne
 	}
 }
 
+/*
+ * defaultlogin_ok -- returns 1 if it's okay to do default-login when:
+ *   there was a login name passed with the --defaultlogin option; and
+ *   the defaultlogin_name contains only "nice" characters;
+ * return 0 otherwise.
+ */
+static int defaultlogin_ok(void)
+{
+       char c, *cp;
+
+       /* An all-alphanumeric defaultlogin name must be supplied. */
+       if (defaultlogin_name == NULL || defaultlogin_name[0] == '\0')
+               return 0;
+       for (cp = defaultlogin_name; (c = *cp); cp++)
+               if (!isalnum(c) && c != '_')
+                       return 0;
+
+       /* All tests are okay, so grant the defaultlogin request. */
+       return 1;
+}
+
 /* do_prompt - show login prompt, optionally preceded by /etc/issue contents */
 static void do_prompt (int showlogin)
 {
@@ -324,8 +347,13 @@ static void do_prompt (int showlogin)
 	}
 	if (nohostname == 0)
 		printf ("%s ", hn);
-	if (showlogin)
-		printf ("login: ");
+	if (showlogin) {
+		printf ("login");
+		if (defaultlogin_ok())
+			printf (" (%s)", defaultlogin_name);
+		printf (": ");
+	}
+
 	fflush (stdout);
 }
 
@@ -378,6 +406,8 @@ static char *get_logname (void)
 
 			*bp++ = c;
 		}
+		if (defaultlogin_ok())
+			break;
 	}
 
 	if (!ascii && (ic = iconv_open("WCHAR_T", "UTF-8"))) {
@@ -409,7 +439,7 @@ static void usage (void)
 		"[--nohangup] [--nohostname] [--long-hostname] "
 		"[--loginprog=/bin/login] [--nice=10] [--delay=10] "
 		"[--chdir=/home] [--chroot=/chroot] [--autologin=user] "
-		"[--loginpause] [--mono] "
+		"[--loginpause] [--mono] [--defaultlogin=user] "
 		"tty' with e.g. tty=tty1", progname);
 }
 
@@ -418,6 +448,7 @@ static struct option const long_options[
 	{ "loginpause", no_argument, &loginpause, 'p' },
 	{ "chdir", required_argument, NULL, 'w' },
 	{ "chroot", required_argument, NULL, 'r' },
+	{ "defaultlogin", required_argument, NULL, 'u'},
 	{ "delay", required_argument, NULL, 'd' },
 	{ "noclear", no_argument, &noclear, 1 },
 	{ "nonewline", no_argument, &nonewline, 1 },
@@ -474,6 +505,9 @@ int main (int argc, char **argv)
 		case 'r':
 			ch_root = optarg;
 			break;
+		case 'u':
+			defaultlogin_name = optarg;
+			break;
 		case 'w':
 			ch_dir = optarg;
 			break;
@@ -498,9 +532,15 @@ int main (int argc, char **argv)
 		do_prompt (0);
 		printf ("login: %s (automatic login)\n", autologin);
 		logname = autologin;
-	} else
-		while ((logname = get_logname ()) == 0)
-			/* do nothing */ ;
+	} else {
+		if (defaultlogin_ok()) {
+			logname = get_logname ();
+			if(*logname == 0)
+				logname = defaultlogin_name;
+		} else
+			while ((logname = get_logname ()) == 0)
+				/* do nothing */ ;
+	}
 
 	if (ch_root)
 		chroot (ch_root);
