diff -urNp -x '*.orig' mingetty-1.08.org/mingetty.8 mingetty-1.08/mingetty.8
--- mingetty-1.08.org/mingetty.8	2008-01-18 12:09:02.000000000 +0100
+++ mingetty-1.08/mingetty.8	2021-05-09 20:15:48.285374609 +0200
@@ -6,7 +6,7 @@ mingetty \- minimal getty for consoles
 [\-\-noclear] [\-\-nonewline] [\-\-noissue] [\-\-nohangup] [\-\-nohostname]
 [\-\-long\-hostname] [\-\-loginprog=/bin/login] [\-\-nice=10] [\-\-delay=5]
 [\-\-chdir=/home] [\-\-chroot=/chroot] [\-\-autologin username]
-[\-\-loginpause]
+[\-\-loginpause] [\-\-mono]
 .I tty
 .PP
 .SH DESCRIPTION
@@ -46,6 +46,9 @@ With this option enabled, the full text
 .B \-\-loginprog /bin/login
 Change the login app.
 .TP
+.B \-\-mono
+Set terminal type to "linux-m" instead of default "linux" (for mono consoles)
+.TP
 .B \-\-nice 10
 Change the priority by calling nice().
 .TP
diff -urNp -x '*.orig' mingetty-1.08.org/mingetty.c mingetty-1.08/mingetty.c
--- mingetty-1.08.org/mingetty.c	2021-05-09 20:15:48.249374609 +0200
+++ mingetty-1.08/mingetty.c	2021-05-09 20:15:48.285374609 +0200
@@ -96,6 +96,8 @@ static char *autologin = NULL;
 static int loginpause = 0;
 /* terminal mode */
 static int mode = K_RAW;
+/* Set mono terminal type */
+static int mono_term = 0;
 
 /* error() - output error messages */
 static void error (const char *fmt, ...)
@@ -407,7 +409,7 @@ static void usage (void)
 		"[--nohangup] [--nohostname] [--long-hostname] "
 		"[--loginprog=/bin/login] [--nice=10] [--delay=10] "
 		"[--chdir=/home] [--chroot=/chroot] [--autologin=user] "
-		"[--loginpause] "
+		"[--loginpause] [--mono] "
 		"tty' with e.g. tty=tty1", progname);
 }
 
@@ -425,6 +427,7 @@ static struct option const long_options[
 	{ "nohostname", no_argument, &nohostname, 1 },
 	{ "loginprog", required_argument, NULL, 'l' },
 	{ "long-hostname", no_argument, &longhostname, 1 },
+	{ "mono", no_argument, &mono_term, 1},
 	{ "nice", required_argument, NULL, 'n' },
 	{ 0, 0, 0, 0 }
 };
@@ -445,7 +448,10 @@ int main (int argc, char **argv)
 #if	defined(s390) || defined(__s390__)
 	putenv ("TERM=dumb");
 #else
-	putenv ("TERM=linux");
+	if (mono_term)
+		putenv ("TERM=linux-m");
+	else
+		putenv ("TERM=linux");
 #endif
 
 	while ((c = getopt_long (argc, argv, "a:p:d:l:n:w:r:", long_options,
