diff -urN mingetty-1.07.org/mingetty.8 mingetty-1.07/mingetty.8
--- mingetty-1.07.org/mingetty.8	2003-05-14 10:55:43.000000000 +0200
+++ mingetty-1.07/mingetty.8	2007-02-11 18:59:03.438623000 +0100
@@ -5,7 +5,7 @@
 .B mingetty
 [\-\-noclear] [\-\-nonewline] [\-\-noissue] [\-\-nohangup] [\-\-nohostname]
 [\-\-long\-hostname] [\-\-loginprog=/bin/login] [\-\-nice=10] [\-\-delay=5]
-[\-\-chdir=/home] [\-\-chroot=/chroot] [\-\-autologin username]
+[\-\-chdir=/home] [\-\-chroot=/chroot] [\-\-autologin username] [\-\-mono]
 .I tty
 .PP
 .SH DESCRIPTION
@@ -45,6 +45,9 @@
 .B \-\-loginprog /bin/login
 Change the login app.
 .TP
+.B \-\-mono
+Set terminal type to "linux-m" instead of default "linux" (for mono consoles)
+.TP
 .B \-\-nice 10
 Change the priority by calling nice().
 .TP
diff -urN mingetty-1.07.org/mingetty.c mingetty-1.07/mingetty.c
--- mingetty-1.07.org/mingetty.c	2007-02-11 18:58:27.658623000 +0100
+++ mingetty-1.07/mingetty.c	2007-02-11 18:58:43.434623000 +0100
@@ -94,6 +94,8 @@
 static char *autologin = NULL;
 /* terminal mode */
 static int mode = K_RAW;
+/* Set mono terminal type */
+static int mono_term = 0;
 
 /* error() - output error messages */
 static void error (const char *fmt, ...)
@@ -401,6 +403,7 @@
 		"[--nohangup] [--nohostname] [--long-hostname] "
 		"[--loginprog=/bin/login] [--nice=10] [--delay=10] "
 		"[--chdir=/home] [--chroot=/chroot] [--autologin=user] "
+		"[--mono] "
 		"tty' with e.g. tty=tty1", progname);
 }
 
@@ -417,6 +420,7 @@
 	{ "nohostname", no_argument, &nohostname, 1 },
 	{ "loginprog", required_argument, NULL, 'l' },
 	{ "long-hostname", no_argument, &longhostname, 1 },
+	{ "mono", no_argument, &mono_term, 1},
 	{ "nice", required_argument, NULL, 'n' },
 	{ 0, 0, 0, 0 }
 };
@@ -437,7 +441,10 @@
 #if	defined(s390) || defined(__s390__)
 	putenv ("TERM=dumb");
 #else
-	putenv ("TERM=linux");
+	if (mono_term)
+		putenv ("TERM=linux-m");
+	else
+		putenv ("TERM=linux");
 #endif
 
 	while ((c = getopt_long (argc, argv, "a:d:l:n:w:r:", long_options,
