diff -ur mingetty-1.07/mingetty.8 mingetty-1.07-remote/mingetty.8
--- mingetty-1.07/mingetty.8	2007-02-11 20:48:29.000000000 +0100
+++ mingetty-1.07-remote/mingetty.8	2007-02-11 20:48:14.000000000 +0100
@@ -65,6 +65,9 @@
 a login name and password. Check the \-f option from
 .B /bin/login
 for this.
+.TP
+.B \-\-remote-host=<hostname>
+Login to remote host. Uses /usr/bin/ssh as default login program.
 .PP
 .SH "ISSUE ESCAPES"
 .B mingetty 
diff -ur mingetty-1.07/mingetty.c mingetty-1.07-remote/mingetty.c
--- mingetty-1.07/mingetty.c	2007-02-11 20:48:29.000000000 +0100
+++ mingetty-1.07-remote/mingetty.c	2007-02-11 20:47:45.000000000 +0100
@@ -50,6 +50,10 @@
 #include <sys/kd.h>
 #include <sys/ttydefaults.h>
 
+#ifndef _PATH_SSH
+#define _PATH_SSH "/usr/bin/ssh"
+#endif
+
 #ifndef IUTF8
 # ifndef ASM_IUTF8
 #  error ASM_IUTF8 input flag not defined - Cannot define IUTF8
@@ -94,6 +98,10 @@
 static int mono_term = 0;
 /* If no login supplied, attempt a default login with this username. */
 static char *defaultlogin_name = NULL;
+/* Log onto remote host */
+static int remote_login = 0;
+/* Remote hostname */
+static char remote_host[MAXHOSTNAMELEN + 1];
 
 /* error() - output error messages */
 static void error (const char *fmt, ...)
@@ -429,7 +439,7 @@
 		"[--nohangup] [--nohostname] [--long-hostname] "
 		"[--loginprog=/bin/login] [--nice=10] [--delay=10] "
 		"[--chdir=/home] [--chroot=/chroot] [--autologin=user] "
-		"[--mono] [--defaultlogin=user] "
+		"[--mono] [--defaultlogin=user] [--remote-host=host] "
 		"tty' with e.g. tty=tty1", progname);
 }
 
@@ -447,20 +458,20 @@
 	{ "loginprog", required_argument, NULL, 'l' },
 	{ "long-hostname", no_argument, &longhostname, 1 },
 	{ "nice", required_argument, NULL, 'n' },
+	{ "remote-host", required_argument, NULL, 2},
 	{ 0, 0, 0, 0 }
 };
 
 int main (int argc, char **argv)
 {
 	char *logname, *s;
+	int option_index = 0;
 	int c;
 
 	progname = argv[0];
 	if (!progname)
 		progname = "mingetty";
 	uname (&uts);
-	gethostname (hn, MAXHOSTNAMELEN);
-	hn[MAXHOSTNAMELEN] = '\0';
 	pid = getpid ();
 	sid = getsid (0);
 #if	defined(s390) || defined(__s390__)
@@ -470,10 +481,15 @@
 #endif
 
 	while ((c = getopt_long (argc, argv, "a:d:l:n:w:r:", long_options,
-		(int *) 0)) != EOF) {
+		&option_index)) != EOF) {
 		switch (c) {
 		case 0:
 			break;
+		case 2:
+			remote_login = 1;
+			strncpy(remote_host, optarg, MAXHOSTNAMELEN);
+			remote_host[MAXHOSTNAMELEN] = '\0';
+			break;
 		case 'a':
 			autologin = optarg;
 			break;
@@ -499,6 +515,13 @@
 			usage ();
 		}
 	}
+	if (!remote_login)
+		  gethostname (hn, MAXHOSTNAMELEN);
+	else {
+		  strncpy(hn, remote_host, MAXHOSTNAMELEN);
+	}
+	hn[MAXHOSTNAMELEN] = '\0';
+
 	if (longhostname == 0 && (s = strchr (hn, '.')))
 		*s = '\0';
 	tty = argv[optind];
@@ -533,7 +556,11 @@
 	if (priority)
 		nice (priority);
 
-	execl (loginprog, loginprog, autologin? "-f" : "--", logname, NULL);
+	if (remote_login) {
+		execl (loginprog, loginprog, "-l", logname, remote_host, NULL);
+	} else {
+		execl (loginprog, loginprog, autologin? "-f" : "--", logname, NULL);
+	}
 	error ("%s: can't exec %s: %s", tty, loginprog, strerror (errno));
 	sleep (5);
 	exit (EXIT_FAILURE);
