diff -ur mingetty-1.07/mingetty.8 mingetty-1.07-remote/mingetty.8
--- mingetty-1.07/mingetty.8	2007-02-11 21:00:04.000000000 +0100
+++ mingetty-1.07-remote/mingetty.8	2007-02-11 20:58:28.000000000 +0100
@@ -69,6 +69,9 @@
 Wait for any key before dropping to the login prompt.
 Can be combined with \fB\-\-autologin\fR to save memory by lazily spawning
 shells.
+.TP
+.B \-\-remote-host=<hostname>
+Login to remote host. Uses /usr/bin/ssh as default login program.
 .PP
 .SH "ISSUE ESCAPES"
 .B mingetty 
diff -ur mingetty-1.07/mingetty.c mingetty-1.07-remote/mingetty.c
--- mingetty-1.07/mingetty.c	2007-02-11 21:00:04.000000000 +0100
+++ mingetty-1.07-remote/mingetty.c	2007-02-11 20:59:44.000000000 +0100
@@ -50,6 +50,10 @@
 #include <sys/kd.h>
 #include <sys/ttydefaults.h>
 
+#ifndef _PATH_SSH
+#define _PATH_SSH "/usr/bin/ssh"
+#endif
+
 #ifndef IUTF8
 # ifndef ASM_IUTF8
 #  error ASM_IUTF8 input flag not defined - Cannot define IUTF8
@@ -98,6 +102,12 @@
 static int mono_term = 0;
 /* If no login supplied, attempt a default login with this username. */
 static char *defaultlogin_name = NULL;
+/* Log onto remote host */
+static int remote_login = 0;
+/* Remote hostname */
+static char remote_host[MAXHOSTNAMELEN + 1];
+/* Do we use user's login app? */
+static int another_login = 0;
 
 /* error() - output error messages */
 static void error (const char *fmt, ...)
@@ -433,7 +443,7 @@
 		"[--nohangup] [--nohostname] [--long-hostname] "
 		"[--loginprog=/bin/login] [--nice=10] [--delay=10] "
 		"[--chdir=/home] [--chroot=/chroot] [--autologin=user] "
-		"[--loginpause] [--mono] [--defaultlogin=user] "
+		"[--loginpause] [--mono] [--defaultlogin=user] [--remote-host=host] "
 		"tty' with e.g. tty=tty1", progname);
 }
 
@@ -453,20 +463,20 @@
 	{ "long-hostname", no_argument, &longhostname, 1 },
 	{ "mono", no_argument, &mono_term, 1},
 	{ "nice", required_argument, NULL, 'n' },
+	{ "remote-host", required_argument, NULL, 2},
 	{ 0, 0, 0, 0 }
 };
 
 int main (int argc, char **argv)
 {
 	char *logname, *s;
+	int option_index = 0;
 	int c;
 
 	progname = argv[0];
 	if (!progname)
 		progname = "mingetty";
 	uname (&uts);
-	gethostname (hn, MAXHOSTNAMELEN);
-	hn[MAXHOSTNAMELEN] = '\0';
 	pid = getpid ();
 	sid = getsid (0);
 #if	defined(s390) || defined(__s390__)
@@ -479,10 +489,17 @@
 #endif
 
 	while ((c = getopt_long (argc, argv, "a:p:d:l:n:w:r:", long_options,
-		(int *) 0)) != EOF) {
+		&option_index)) != EOF) {
 		switch (c) {
 		case 0:
 			break;
+		case 2:
+			remote_login = 1;
+			strncpy(remote_host, optarg, MAXHOSTNAMELEN);
+			remote_host[MAXHOSTNAMELEN] = '\0';
+			if (another_login == 0)
+				loginprog = strdup(_PATH_SSH);
+			break;
 		case 'a':
 			autologin = optarg;
 			break;
@@ -490,6 +507,7 @@
 			delay = atoi (optarg);
 			break;
 		case 'l':
+			another_login = 1;
 			loginprog = optarg;
 			break;
 		case 'n':
@@ -508,6 +526,13 @@
 			usage ();
 		}
 	}
+	if (!remote_login)
+		  gethostname (hn, MAXHOSTNAMELEN);
+	else {
+		  strncpy(hn, remote_host, MAXHOSTNAMELEN);
+	}
+	hn[MAXHOSTNAMELEN] = '\0';
+
 	if (longhostname == 0 && (s = strchr (hn, '.')))
 		*s = '\0';
 	tty = argv[optind];
@@ -542,7 +567,11 @@
 	if (priority)
 		nice (priority);
 
-	execl (loginprog, loginprog, autologin? "-f" : "--", logname, NULL);
+	if (remote_login) {
+		execl (loginprog, loginprog, "-l", logname, remote_host, NULL);
+	} else {
+		execl (loginprog, loginprog, autologin? "-f" : "--", logname, NULL);
+	}
 	error ("%s: can't exec %s: %s", tty, loginprog, strerror (errno));
 	sleep (5);
 	exit (EXIT_FAILURE);
